tests/kernel/mutex/mutex_null_ptr_deref/src/main.c

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2022 SeediqQ from Issue 48056
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/ztest.h>
+
+#define TIMEOUT_MS 100
+#define STACKSZ	   8192
+
+static struct k_mutex mutex;
+static struct k_thread thread;
+
+K_THREAD_STACK_DEFINE(stack, STACKSZ);
+
+static void test_thread(void *p1, void *p2, void *p3)
+{
+	ARG_UNUSED(p1);
+	ARG_UNUSED(p2);
+	ARG_UNUSED(p3);
+
+	k_mutex_lock(&mutex, K_MSEC(TIMEOUT_MS));
+
+	k_mutex_unlock(&mutex);
```

Comment:
I think that we should be verifying that k_mutex_lock() fails as it is expected to timeout (which means that k_mutex_unlock is unnecessary).